src/libsystemd/sd-id128/id128-util.c

```diff
@@ -102,6 +103,21 @@ int id128_read_fd(int fd, Id128Flag f, sd_id128_t *ret) {
         return r == -EINVAL ? -EUCLEAN : r;
 }
 
+int id128_read_at(int dir_fd, const char *path, Id128Flag f, sd_id128_t *ret) {
```

Comment:
This function should not do the symlink chasing. We should leave that to `id128_get_machine_at()`. This one should use `xopenat()` and then work on the file descriptor returned by that.

Reply:
OK. You mean you do not like 830e52caa2bf1a29f56cb93e7ed85acb1bda11c3. I will revert it and rework in that way.